File: 2021/py/day02.py
from aocd import get_data


def parse_data():
    raw_data = get_data(year=2021, day=2)
    data = [line.split(" ") for line in raw_data.splitlines()]
    return data


def part_one(data):
    position = 0
    depth = 0

    for cmd, amt in data:
        if cmd == 'forward':
            position += int(amt)
        if cmd == 'down':
            depth += int(amt)
        if cmd == 'up':
            depth -= int(amt)

    return position * depth


def part_two(data):
    position = 0
    depth = 0
    aim = 0

    for cmd, amount in data:
        amt = int(amount)
        if cmd == 'down':
            aim += amt
        if cmd == 'up':
            aim -= amt
        if cmd == 'forward':
            position += amt
            depth += aim * amt

    return position * depth


if __name__ == "__main__":
    data = parse_data()
    print("*" * 10, "Part One", "*" * 10)
    print(part_one(data))
    print("*" * 10, "Part Two", "*" * 10)
    print(part_two(data))

File: 2021/golang/day02/day02.go
package main

import (
	"fmt"
	utils "raybuhr/aoc/utils"
	"strconv"
	"strings"
)

func main() {
	var part1 int
	var part2 int
	data := utils.ReadLines("data.txt")
	part1 = moveTwoD(data)
	part2 = moveAndAim(data)
	fmt.Println("Part 1:", part1)
	fmt.Println("Part 2:", part2)
}

func moveTwoD(data []string) int {
	position := 0
	depth := 0
	for _, line := range data {
		step := strings.Split(line, " ")
		cmd := step[0]
		amt, _ := strconv.Atoi(step[1])
		switch {
		case cmd == "forward":
			position += amt
		case cmd == "down":
			depth += amt
		case cmd == "up":
			depth -= amt
		}
	}
	return position * depth
}

func moveAndAim(data []string) int {
	position := 0
	depth := 0
	aim := 0

	for _, line := range data {
		step := strings.Split(line, " ")
		cmd := step[0]
		amt, _ := strconv.Atoi(step[1])
		switch {
		case cmd == "forward":
			position += amt
			depth += aim * amt
		case cmd == "down":
			aim += amt
		case cmd == "up":
			aim -= amt
		}
	}
	return position * depth
}
